[leveldb] build failure

**Host Environment**

- OS: Debian testing
- Compiler: `g++ 12.2.0`

<details>
<summary>Compiler versions</summary>

```
$ /usr/bin/c++ --version                                                             
c++ (Debian 12.2.0-3) 12.2.0
Copyright (C) 2022 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

$ ls -l /etc/alternatives/c++
lrwxrwxrwx 1 root root 12 Oct 10  2021 /etc/alternatives/c++ -> /usr/bin/g++
```

</details>

**To Reproduce**

Steps to reproduce the behavior:

1. Install the latest `vcpkg`:
```
   $ git clone https://github.com/Microsoft/vcpkg.git /path/to/vcpkg
   $ cd /path/to/vcpkg && ./bootstrap-vcpkg.sh
   $ cd /path/to/scilla
   $ export VCPKG_ROOT=/path/to/vcpkg
```
2. Create a directory with the `vcpkg.json` file that contains:
```json
{
  "name": "test",
  "version-string": "1.0.0",
  "dependencies": [
    {
      "name": "leveldb",
      "default-features": false,
      "features": [
        "crc32c",
        "snappy"
      ]
    }
  ],
  "overrides": []
}
```
3.  Run the following command in the directory:  `$VCPKG_ROOT/vcpkg install --triplet x64-linux-dynamic`

**Failure logs**

<details>
<summary>`vcpkg install` errors</summary>

```
Detecting compiler hash for triplet x64-linux...
Detecting compiler hash for triplet x64-linux-dynamic...
The following packages will be built and installed:
    leveldb[core,crc32c,snappy]:x64-linux-dynamic -> 1.22#5
Restored 0 package(s) from /home/jubnzv/.cache/vcpkg/archives in 4.445 us. Use --debug to see more details.
Installing 1/1 leveldb:x64-linux-dynamic...
Building leveldb[core,crc32c,snappy]:x64-linux-dynamic...
warning: -- Using community triplet x64-linux-dynamic. This triplet configuration is not guaranteed to succeed.
-- [COMMUNITY] Loading triplet configuration from: /home/jubnzv/Sources/vcpkg/triplets/community/x64-linux-dynamic.cmake
-- Note: leveldb only supports static library linkage. Building static library.
-- Using cached google-leveldb-1.22.tar.gz.
-- Cleaning sources at /home/jubnzv/Sources/vcpkg/buildtrees/leveldb/src/1.22-3348a839cb.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /home/jubnzv/Sources/vcpkg/downloads/google-leveldb-1.22.tar.gz
-- Applying patch fix_config.patch
-- Applying patch fix-dependencies.patch
-- Using source at /home/jubnzv/Sources/vcpkg/buildtrees/leveldb/src/1.22-3348a839cb.clean
-- Found external ninja('1.11.1').
-- Configuring x64-linux-dynamic
-- Building x64-linux-dynamic-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:131 (message):
    Command failed: /usr/bin/cmake --build . --config Debug --target install -- -v -j9
    Working Directory: /home/jubnzv/Sources/vcpkg/buildtrees/leveldb/x64-linux-dynamic-dbg
    See logs for more information:
      /home/jubnzv/Sources/vcpkg/buildtrees/leveldb/install-x64-linux-dynamic-dbg-out.log

Call Stack (most recent call first):
  /home/jubnzv/Dev/vcpkg-bug/vcpkg_installed/x64-linux/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  /home/jubnzv/Dev/vcpkg-bug/vcpkg_installed/x64-linux/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  ports/leveldb/portfile.cmake:28 (vcpkg_cmake_install)
  scripts/ports.cmake:147 (include)


error: building leveldb:x64-linux-dynamic failed with: BUILD_FAILED
error: Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
    https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+leveldb
You can submit a new issue at:
    https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[leveldb]+Build+error
Include '[leveldb] Build error' in your bug report title, the following version information in your bug description, and attach any relevant failure logs from above.
    vcpkg-tool version: 2022-10-17-3247920fbdd47d08f36cbd480addd9890d3c2435
    vcpkg-scripts version: 7ad236f60 2022-10-24 (11 hours ago)


Please use the prefilled template from /home/jubnzv/Dev/vcpkg-bug/vcpkg_installed/vcpkg/issue_body.md when reporting your issue.
```

</details>

* [buildtrees/leveldb/install-x64-linux-dynamic-dbg-out.log](https://gist.github.com/jubnzv/77514a8055c51b3d7a34ca072ee9644a)

--- a/ports/leveldb/portfile.cmake
+++ b/ports/leveldb/portfile.cmake
@@ -2,7 +2,7 @@ vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
 
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
-    REPO "google/leveldb"
+    REPO google/leveldb
     REF 1.22
     SHA512 f9bbf5f466e7f707b94e19261762319ea9f65d41911690e84f59098551e2e69beccf756a414d705ade74ee96fd979bdb8b94c171c6f2cc83873cbd4a9380dbab
     HEAD_REF master
@@ -37,6 +37,4 @@ vcpkg_cmake_config_fixup(CONFIG_PATH "lib/cmake/${PORT}")
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
 
-# Handle copyright
 file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
-

--- a/ports/leveldb/portfile.cmake
+++ b/ports/leveldb/portfile.cmake
@@ -4,7 +4,7 @@ vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO "google/leveldb"
     REF 1.22
-    SHA512  f9bbf5f466e7f707b94e19261762319ea9f65d41911690e84f59098551e2e69beccf756a414d705ade74ee96fd979bdb8b94c171c6f2cc83873cbd4a9380dbab
+    SHA512 f9bbf5f466e7f707b94e19261762319ea9f65d41911690e84f59098551e2e69beccf756a414d705ade74ee96fd979bdb8b94c171c6f2cc83873cbd4a9380dbab
     HEAD_REF master
     PATCHES
         fix_config.patch
@@ -15,19 +15,23 @@ file(COPY "${CURRENT_PORT_DIR}/leveldbConfig.cmake.in" DESTINATION "${SOURCE_PAT
 
 vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
     FEATURES
-        crc32c  WITH_CRC32C
-        snappy  WITH_SNAPPY
+        crc32c WITH_CRC32C
+        snappy WITH_SNAPPY
 )
 
 vcpkg_cmake_configure(
     SOURCE_PATH "${SOURCE_PATH}"
-    OPTIONS ${FEATURE_OPTIONS}
-    OPTIONS_DEBUG -DINSTALL_HEADERS=OFF
+    OPTIONS
+        ${FEATURE_OPTIONS}
+        -DLEVELDB_BUILD_TESTS=OFF
+        -DLEVELDB_BUILD_BENCHMARKS=OFF
+    OPTIONS_DEBUG
+        -DINSTALL_HEADERS=OFF
 )
 
 vcpkg_cmake_install()
 
-vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/${PORT})
+vcpkg_cmake_config_fixup(CONFIG_PATH "lib/cmake/${PORT}")
 
 
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")

--- a/ports/leveldb/vcpkg.json
+++ b/ports/leveldb/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "leveldb",
   "version": "1.22",
-  "port-version": 5,
+  "port-version": 6,
   "description": "LevelDB is a fast key-value storage library written at Google that provides an ordered mapping from string keys to string values.",
   "homepage": "https://github.com/google/leveldb",
   "license": "BSD-3-Clause",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -3530,7 +3530,7 @@
     },
     "leveldb": {
       "baseline": "1.22",
-      "port-version": 5
+      "port-version": 6
     },
     "levmar": {
       "baseline": "2.6",

--- a/versions/l-/leveldb.json
+++ b/versions/l-/leveldb.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "9f2027d4e52e27c17c8de783f2a2c3b8190b9082",
+      "version": "1.22",
+      "port-version": 6
+    },
+    {
       "git-tree": "2ca11f66c6efa417b28755759b6c67670524f532",
       "version": "1.22",
       "port-version": 5

--- a/versions/l-/leveldb.json
+++ b/versions/l-/leveldb.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "9f2027d4e52e27c17c8de783f2a2c3b8190b9082",
+      "git-tree": "c7a1016e970ae104981e8a8fcc5069593bf716b3",
       "version": "1.22",
       "port-version": 6
     },

